### PR TITLE
feat: Implement new window requested event, closes #527

### LINKED ISSUE #527: Add new window requested event callback/event
**Is your feature request related to a problem? Please describe.**
I'd like to be able to intercept and view information about new window events (ie: popups, target=_blank links, etc) and choose whether to allow them.

**Describe the solution you'd like**
A way to register a handler for new window events.

**Describe alternatives you've considered**
Onbeforeload javascript to rewrite and/or register javascript event handlers on all potential sources of navigation - this does not cover programatically added buttons and a number of other cases. 
Proxying - requires a local proxy on the entire host machine OR implementation of the proxying configuration on webviews. Either option would probably be quite cumbersome.

**Would you assign yourself to implement this feature?**
- [x] Yes
- [ ] No

**Additional context**
I'm creating this issue as more of a formality than anything - I already have a PR for this open

### PR DESCRIPTION
**Pull request by atlanticaccent**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [ ] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### Other information

The WKWebview implementation is outstanding but I'm giving it a shot.
Also couldn't test this properly as whilst testing using WSL I couldn't get popup windows to work.

Regardless of the above, these "naïve" implementations may allow for new popup windows to bypass/ignore existing webview settings. For example, they do not inherit parent window navigation handlers for example. This isn't a new thing - it's just more noticeable when implementing a feature like this where we have functions that can cancel events.

### CODE REVIEW

File: examples/new_window_req_event.rs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        <a href="https://www.wikipedia.org" target="_blank">Visit Wikipedia</a>
+        <a href="https://www.github.com" target="_blank">(Try to) visit GitHub</a>
+      </div>
+    </body>
+  "#;
+
+  let event_loop: EventLoop<UserEvent> = EventLoop::with_user_event();
+  let proxy = event_loop.create_proxy();
+  let mut webcontext = WebContext::default();
+  webcontext.set_allows_automation(true);
```
Comment (amrbashir) [inline examples/new_window_req_event.rs]:
```suggestion
```

File: src/webview/mod.rs [hunk 2]
```diff
@@ -322,6 +329,18 @@ impl<'a> WebViewBuilder<'a> {
     self
   }
 
+  /// Set a new window request handler to decide if incoming url is allowed to be opened.
+  ///
+  /// The closure takes a `String` parameter as url and return `bool` to determine the url. True is
+  /// allowed to open the url in a new window and false if not.
```
Comment (amrbashir) [inline src/webview/mod.rs]:
```suggestion
  /// allow to open the url in a new window and false is not.
```

**Review by amrbashir — CHANGES_REQUESTED** [review]
Windows, and Linux implementation is LGTM. There is some small nit-picks and I will leave reviewing the macOS to @wusyong

**Review by keiya01 — COMMENTED** [review]
It's working great on macos.
it looks good to me.

**Review by wusyong — DISMISSED** [review]

**Review by amrbashir — DISMISSED** [review]

**Review by amrbashir — APPROVED** [review]

**Pull request merged**